Makefile

```diff
@@ -1,12 +1,18 @@
+# Give Selfie information about the operating system ($OS is defined on Windows only)
+ifndef OS
+	# use uname or fail with "Unknown"
+	OS := $(shell sh -c 'uname 2>/dev/null || echo Unknown')
+endif
+
 # Compiler flags
-CFLAGS := -Wall -Wextra -O3 -m64 -D'uint64_t=unsigned long long'
+CFLAGS := -Wall -Wextra -O3 -m64 -D'uint64_t=unsigned long long' -D'HOST_OS="$(OS)"'
```

Comment:
You can keep that but pass in an integer that identifies the OS. This avoids any string dependencies in selfie.

Reply:
Done